### PR TITLE
GET /api/articles/:article_id  completed with test

### PR DESCRIPTION
**Pull request by Kingsl8yer**
Completed GET /api/articles/:article_id with its tests

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -53,3 +53,33 @@ describe("GET /api", () => {
...
+  test("status 200: responds with an article object", () => {
+      return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+          expect(body.article.length).toBe(1);
+          expect(typeof body.article[0].author).toBe("string");
+          expect(typeof body.article[0].title).toBe("string");
+          expect(typeof body.article[0].article_id).toBe("number");
+          expect(typeof body.article[0].body).toBe("string");
```
Comment (pippaamy) [inline __tests__/index.test.js]: we can expect the article id number to be 1

File: __tests__/index.test.js [hunk 2]
```diff
@@ -53,3 +53,33 @@ describe("GET /api", () => {
...
+  //Error handling
+  test("status 404: responds with error message when given a wrong URL", () => {
+      return request(app)
+      .get("/api/articles/1000")
+      .expect(404)
+      .then(({ body }) => {
+          expect(body.msg).toBe("Article not found");
+      });
+  });
+});
```
Comment (pippaamy) [inline __tests__/index.test.js:94]: what other error could we have here?

**Review by pippaamy — COMMENTED** [review]
good job, just a line to change on your tests and one more test to add :)

File: MVC/app.js [hunk 3]
```diff
@@ -1,19 +1,32 @@
...
+    if(err.code === '22P02')
+    res.status(400).send({msg: "Bad request"})
+    next(err);
+  })
 
-app.all('/*', (req, res, next) => {
-    res.status(404).send({msg: 'URL not found'});
+app.use((err, req, res, next) => {
+    if(err.status) res.status(err.status).send({msg: err.msg});
+    else next(err);
```
Comment (B7Bprog) [inline MVC/app.js]: Here you are calling next to jump to the next error handling middleware, but have you got a next one? Please write one to handle internal server errors. 
Please check out the end of this page in the notes here:
https://notes.northcoders.com/courses/js-back-end/error-handling

File: MVC/models/articlesModels.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const db = require('../../db/connection');
+
+exports.selectArticleById = (article_id) => {
+    const sql = 'SELECT * FROM articles WHERE article_id = $1';
+    return db.query(sql, [article_id]).then(({rows}) => {
+        if (rows.length === 0) {
+            return Promise.reject({status: 404, msg: 'Article not found'});
+        } else {
+            return rows;
```
Comment (B7Bprog) [inline MVC/models/articlesModels.js]: Please make sure you are sending back an object, not the array containing the object here. How could you extract that object from this 'rows' array here?

File: MVC/controllers/articlesController.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const {selectArticleById} = require('../models/articlesModels');
+
+exports.getArticleById = (req, res, next) => {
+    const {article_id} = req.params;
+    selectArticleById(article_id)
+    .then(article => {
+        res.status(200).send({article});
```
Comment (B7Bprog) [inline MVC/controllers/articlesController.js:7]: Here you should respond with an object that has a key of 'article' and the value is not an array, but a single object. (After refactoring the model, it should be fine.)

File: __tests__/index.test.js [hunk 6]
```diff
@@ -53,3 +53,44 @@ describe("GET /api", () => {
...
 });
+
+describe("GET /api/articles/:article_id", () => {
+  test("status 200: responds with an article object", () => {
+      return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+          expect(body.article.length).toBe(1);
+          expect(body.article[0].article_id).toBe(1);
```
Comment (B7Bprog) [inline __tests__/index.test.js]: Here body.article should be an object, not an array.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Miguel, nice work refactoring your tests. Please see my comment and refine your code, then when it's done and the tests are passing, please merge.

**Pull request merged**